recipes/hdf5/meta.yaml

```diff
@@ -0,0 +1,39 @@
+{% set version = "1.8.16" %}
+
+package:
+  name: hdf5
+  version: {{ version }}
+
+source:
+  fn: hdf5-{{version}}.tar.gz
+  url: http://www.hdfgroup.org/ftp/HDF5/current/src/hdf5-{{version}}.tar.gz
+  md5: b8ed9a36ae142317f88b0c7ef4b9c618
+  patches:
+    - test_Makefile.in.patch
```

Comment:
I'd want a really good comment in here to explain the rationale. Maybe provide a link back to the original PR too.

P.S. I'm happy for this to be done in the feedstock once merged.